robots/commenter/main.go

```diff
@@ -209,13 +237,84 @@ func makeCommenter(comment string, useTemplate bool) func(meta) (string, error)
 	}
 }
 
-func run(c client, query, sort string, asc, random bool, commenter func(meta) (string, error), ceiling int) error {
-	log.Printf("Searching: %s", query)
-	issues, err := c.FindIssues(query, sort, asc)
+func getLastValidLabelsUpdate(ctx context.Context, client *github2.Client, issue github.Issue, goodLabels map[string]bool, badLabels map[string]bool) (time.Time, error) {
+	org, repo, number, err := parseHTMLURL(issue.HTMLURL)
 	if err != nil {
-		return fmt.Errorf("search failed: %v", err)
+		return time.Time{}, fmt.Errorf("error parsing HTML of PR %d: %v", issue.ID, err)
+	}
+	readyDate := time.Time{}
+	var foundLabel bool
+	for i := 1; true; i++ {
+		prTimeline, _, err := client.Issues.ListIssueTimeline(ctx, org, repo, number, &github2.ListOptions{PerPage: 100, Page: i})
+		if err != nil {
+			return time.Time{}, fmt.Errorf("could not get PR timeline PR %d in repo %s/%s: %v", issue.ID, org, repo, err)
+		}
+		// break loop if there is no timeline entry in PR
+		if len(prTimeline) == 0 {
+			break
+		}
+		// loop all events, search for labels and check the create date for good labels
+		for _, v := range prTimeline {
+			if v.Label != nil && *v.Event == "labeled" && goodLabels[*v.Label.Name] && v.CreatedAt.After(readyDate) {
+				readyDate = *v.CreatedAt
+				foundLabel = true
+			}
+		}
+		// if a label were found, loop all events again, search for labels and check the create date for bad labels
+		if foundLabel {
+			for _, v := range prTimeline {
+				if v.Label != nil && *v.Event == "unlabeled" && badLabels[*v.Label.Name] && v.CreatedAt.Before(readyDate) {
```

Comment:
```suggestion
				if v.Label != nil && *v.Event == "unlabeled" && badLabels[*v.Label.Name] && v.CreatedAt.After(readyDate) {
```
I think this is also an after

Reply:
You're right... we want to get the last "unlabled" event of the labels.